src/fsharp/Optimizer.fs

```diff
@@ -397,10 +422,17 @@ type cenv =
       
       /// cache methods with SecurityAttribute applied to them, to prevent unnecessary calls to ExistsInEntireHierarchyOfType
       casApplied: Dictionary<Stamp, bool>
+
     }
 
     override x.ToString() = "<cenv>"
 
+// environment for a method
+type menv =
```

Comment:
nit: I think we should rename this to `methEnv` or `MethodEnv` considering there are a lot of `(insert letter)env` types; would make it less ambiguous